### PR TITLE
[Component Library] Table components

### PR DESCRIPTION
**Pull request by dmmulroy**
# [Component Library] Table components
- Adds the following components to the library:
  -  `Table` with `Fixed`, `Responsive`, and `Unstyled` variants
  -  `Table.Head`
  -  `Table.Body`
  -  `Table.Row`
  -  `Table.Header_cell`
  -  `Table.Data_cell`
  
  ## Screenshots
### Table (Light mode)
image

  ### Table (Dark Mode)
image

### CODE REVIEW

**Comment by tjdevries** [comment]
Ok hear me out.... what if a Table was a functor that took some type "t" that then all the children had to send in records of that type so you couldn't forget columns, etc??

Am i crazy?

**Comment by ghost** [comment]
@tjdevries I really like that idea - and is what I was kind of playing around with the `Column_spec.t` stuff in this code snippet

```ocaml
module Column_spec = struct
  type ('data, 'column) t =
    { header : string
    ; accessor : 'data -> 'column
    ; to_string : 'column -> string
    }

  let header { header; _ } = header
  let accessor { accessor; _ } = accessor
  let value { accessor; _ } data = accessor data

  let to_header_cell column =
    column.header |> txt |> List.return |> Header_cell.make
  ;;

  let to_data_cell datum column =
    datum
    |> column.accessor
    |> column.to_string
    |> txt
    |> List.return
    |> Data_cell.make
  ;;
end

module Responsive_table = struct
  let make
    (type data)
    ~(columns : (data, 'a) Column_spec.t list)
    ~(data : data list)
    ()
    =
    let thead =
      columns
      |> List.map ~f:Column_spec.to_header_cell
      |> Row.make
      |> List.return
      |> Head.make
    in
    let tbody =
      data
      |> List.map ~f:(fun datum ->
        columns |> List.map ~f:(Column_spec.to_data_cell datum))
      |> List.map ~f:Row.make
      |> Body.make
      |> List.return
    in
    tablex ~thead tbody
  ;;
end
```


My plan in is to create a `Data_table` or something similar that does what you're describing using the table components in this pr as the foundation. 

I definitely plan on wrapping/composing some of these more primitive/"atom" components to be more opinionated/useful

**Comment by tjdevries** [comment]
Cool, we can leave that for further exploration, i'll consider the components "just for generating html" kind of thing. i'll take a more detailed look

File: lib/components/table.ml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+end
+
+module Body = struct
+  type attributes = Html_types.tbody_attrib Tyxml_html.attrib list
+  type children = Html_types.tbody_content_fun elt list_wrap
+
+  let base_classes = [ "bg-slate-50"; "dark:bg-slate-800" ]
+
+  let make ?(classes = []) ?(attributes = []) children =
+    let attrs = attributes @ [ a_class (base_classes @ classes) ] in
```
Comment (tjdevries) [inline lib/components/table.ml]: i think technically you can avoid `base_classes @ classes` by just doing making `let base_classes = a_class [ "..."]` and then `let attrs = attributes @ [ base_classes; a_class classes ]`

**Pull request merged**